### DISCUSSION TITLE
Band pattern 'QA60' did not match any bands

### DISCUSSION THREAD

**Discussion by eshai90** [community/community#134206] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have an error when trying to download images from the year 2024. I can't download the new versions because I have functions that also give me errors.

Number of images available between 2024-01-01 and 2024-04-09:
- In Landsat Tier 1 & Sentinel-2 Level-1C:
     S2: 43 images
     L9: 16 images
  Total to download: 59 images
- In Landsat Tier 2 (not suitable for time-series analysis):
  Total Tier 2: 0 images

Downloading images:
S2: 43 images
32%
---------------------------------------------------------------------------
HttpError                                 Traceback (most recent call last)
File ~\.conda\envs\ELR5\lib\site-packages\ee\data.py:328, in _execute_cloud_call(call, num_retries)
    327 try:
--> 328   return call.execute(num_retries=num_retries)
    329 except googleapiclient.errors.HttpError as e:

File ~\.conda\envs\ELR5\lib\site-packages\googleapiclient\_helpers.py:134, in positional.<locals>.positional_decorator.<locals>.positional_wrapper(*args, **kwargs)
    133         logger.warning(message)
--> 134 return wrapped(*args, **kwargs)

File ~\.conda\envs\ELR5\lib\site-packages\googleapiclient\http.py:915, in HttpRequest.execute(self, http, num_retries)
    914 if resp.status >= 300:
--> 915     raise HttpError(resp, content, uri=self.uri)
    916 return self.postproc(resp, content)

HttpError: <HttpError 400 when requesting https://earthengine.googleapis.com/v1alpha/projects/earthengine-legacy/value:compute?prettyPrint=false&alt=json returned "Image.select: Band pattern 'QA60' did not match any bands. Available bands: [B1, B2, B3, B4, B5, B6, B7, B8, B8A, B9, B10, B11, B12, MSK_CLASSI_OPAQUE, MSK_CLASSI_CIRRUS, MSK_CLASSI_SNOW_ICE]". Details: "Image.select: Band pattern 'QA60' did not match any bands. Available bands: [B1, B2, B3, B4, B5, B6, B7, B8, B8A, B9, B10, B11, B12, MSK_CLASSI_OPAQUE, MSK_CLASSI_CIRRUS, MSK_CLASSI_SNOW_ICE]">

During handling of the above exception, another exception occurred:

EEException                               Traceback (most recent call last)
Input In [7], in <cell line: 2>()
      1 #4: Download Data (images) 
----> 2 metadata = SDS_download.retrieve_images(inputs)

File ~\Documents\Codigos_MRS\Coastsat_ELR\coastsat\SDS_download.py:326, in retrieve_images(inputs)
    324 ee_region_ms = adjust_polygon(inputs['polygon'],proj_ms)
    325 ee_region_swir = adjust_polygon(inputs['polygon'],proj_swir)
--> 326 ee_region_mask = adjust_polygon(inputs['polygon'],proj_mask)
    327 # download the ms, swir and QA bands from EE
    328 count = 0

File ~\Documents\Codigos_MRS\Coastsat_ELR\coastsat\SDS_download.py:671, in adjust_polygon(polygon, proj)
    669 polygon_ee = ee.Geometry.Polygon(polygon)    
    670 # convert polygon to image coordinates
--> 671 polygon_coords = np.array(ee.List(polygon_ee.transform(proj, 1).coordinates().get(0)).getInfo())
    672 # make it a rectangle
    673 xmin = np.min(polygon_coords[:,0])

File ~\.conda\envs\ELR5\lib\site-packages\ee\computedobject.py:98, in ComputedObject.getInfo(self)
     92 def getInfo(self):
     93   """Fetch and return information about this object.
     94 
     95   Returns:
     96     The object can evaluate to anything.
     97   """
---> 98   return data.computeValue(self)

File ~\.conda\envs\ELR5\lib\site-packages\ee\data.py:738, in computeValue(obj)
    735 if workload_tag:
    736   body['workloadTag'] = workload_tag
--> 738 return _execute_cloud_call(
    739     _get_cloud_api_resource().projects().value().compute(
    740         body=body,
    741         project=_get_projects_path(),
    742         prettyPrint=False))['result']

File ~\.conda\envs\ELR5\lib\site-packages\ee\data.py:330, in _execute_cloud_call(call, num_retries)
    328   return call.execute(num_retries=num_retries)
    329 except googleapiclient.errors.HttpError as e:
--> 330   raise _translate_cloud_exception(e)

EEException: Image.select: Band pattern 'QA60' did not match any bands. Available bands: [B1, B2, B3, B4, B5, B6, B7, B8, B8A, B9, B10, B11, B12, MSK_CLASSI_OPAQUE, MSK_CLASSI_CIRRUS, MSK_CLASSI_SNOW_ICE]

**Comment by LiteBrite82** [comment]
Hi @eshai90 ,

Thanks for being a part of the GitHub Community, we're glad you're here!

It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
